test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/JSSymbolTest.scala

```diff
@@ -304,26 +336,40 @@ object JSSymbolTest {
     def callSymbol(sym: js.Symbol)(args: js.Any*): Any = js.native
   }
 
-  def singletonIterator(singleton: Any): js.Dynamic = {
-    var first = true
+  def singletonIterator(singleton: Any): js.Dynamic =
+    js.Dynamic.literal(next = singletonIteratorLogic(singleton))
+
+  def singletonAsyncIterator(singleton: Any): js.Dynamic = {
+    val logic = singletonIteratorLogic(singleton)
     js.Dynamic.literal(
-        next = { () =>
-          if (first) {
-            first = false
-            js.Dynamic.literal(value = singleton.asInstanceOf[js.Any],
-                done = false)
-          } else {
-            js.Dynamic.literal(value = (), done = true)
-          }
-        }
+      next = { () =>
+        new js.Promise[js.Dynamic]({
+          (resolve: js.Function1[js.Dynamic | js.Thenable[js.Dynamic], _], reject: js.Function1[Any, _]) =>
+            resolve(logic())
+        })
+      }
     )
   }
 
+  def singletonIteratorLogic(singleton: Any): () => js.Dynamic = {
+    var first = true
+    () => {
+      if (first) {
+        first = false
+        js.Dynamic.literal(value = singleton.asInstanceOf[js.Any],
+          done = false)
```

Comment:
Either
```suggestion
        js.Dynamic.literal(value = singleton.asInstanceOf[js.Any],
            done = false)
```
or now on one line:
```suggestion
        js.Dynamic.literal(value = singleton.asInstanceOf[js.Any], done = false)
```